packages/dev/core/src/XR/features/WebXRDepthSensing.ts

```diff
@@ -0,0 +1,320 @@
+import { RawTexture } from "../../Materials/Textures/rawTexture";
+import { WebXRFeatureName, WebXRFeaturesManager } from "../webXRFeaturesManager";
+import type { WebXRSessionManager } from "../webXRSessionManager";
+import { WebXRAbstractFeature } from "./WebXRAbstractFeature";
+import { Tools } from "../../Misc/tools";
+import { Texture } from "../../Materials/Textures/texture";
+import { Engine } from "../../Engines/engine";
+import { Observable } from "../../Misc/observable";
+
+/**
+ * Options for Depth Sensing feature
+ */
+export interface IWebXRDepthSensingOptions {
+    /**
+     *  The desired depth sensing usage for the session
+     */
+    usagePreference: XRDepthUsage[];
+    /**
+     * The desired depth sensing data format for the session
+     */
+    dataFormatPreference: XRDepthDataFormat[];
+}
+
+type GetDepthInMetersType = (x: number, y: number) => number;
+
+/**
+ * WebXR Feature for WebXR Depth Sensing Module
+ */

```

Comment:
```suggestion
 * @since
 */
```

This will automatically populate the version in which it was added.